src/integration.ts

```diff
@@ -0,0 +1,134 @@
+import { CallExpr } from "./expression";
+import { ASL, State } from "./asl";
+import { FunctionlessNode } from "./node";
+import { VTL } from "./vtl";
+import { AnyFunction } from "./util";
+
+/**
+ * Integration types supported by Functionless.
+ *
+ * Add an integration by creating a any object that has a property named "kind" and either implements the
+ * {@link IntegrationHandler} interface or has methods that implement it (or both).
+ *
+ * Example showing both strategies:
+ *
+ * ```ts
+ * export class Function implements IntegrationHandler {
+ *    readonly kind = "Function",
+ *
+ *    // Integration Handler for ASL
+ *    asl(call, context) {
+ *       // return Step Function task.
+ *    }
+ *
+ *    default = makeIntegration<() => string>({
+ *        kind: "Function.default",
+ *        asl: (call, context) => {
+ *            // return step function task
+ *        }
+ *    });
+ * }
+ *
+ * // an interface to provide the actual callable methods to users
+ * export interface Function {
+ *    // call me to send a string payload
+ *    (payload: String) => string
+ * }
+ *
+ * // use
+ *
+ * const func1 = new Function(...);
+ * // uses the ASL
+ * new StepFunction(..., () => {
+ *    func1("some string");
+ *    func1.default();
+ * })
+ * ```
+ *
+ * If an integration does not support an integration type, leave the function undefined or throw an error.
+ *
+ * Implement the unhandledContext function to customize the error message for unsupported contexts.
+ * Otherwise the error will be: `${this.name} is not supported by context ${context.kind}.`
+ */
+export interface IntegrationHandler {
+  /**
+   * Integration Handler kind - for example StepFunction.describeExecution
+   */
+  readonly kind: string;
+  unhandledContext?: <T>(kind: string, context: CallContext) => T;
+  /**
+   * Integrate with VTL applications like AppAsync.
+   * @private
```

Comment:
private? What's this for?

Reply:
It will omit this function from the docs. , We don't want StepFunctions.asl to show up on the docs as it's on the computer side, not the user side.